### PR TITLE
Added dataset handle class

### PR DESCRIPTION
**Pull request by mohami2000**
Added dataset handle class for future integration with datasets

https://b.corp.google.com/issues/313706281

### CODE REVIEW

File: src/kagglehub/handle.py [hunk 1]
```diff
@@ -43,6 +43,28 @@ def to_url(self) -> str:
             return f"https://www.kaggle.com/models/{self.owner}/{self.model}/frameworks/{self.framework}/variations/{self.variation}"
 
 
+@dataclass
+class DatasetHandle:
+    owner: str
+    dataset_name: str
+    version: Optional[int] = None
+
+    def is_versioned(self) -> bool:
```
Comment (neshdev) [inline src/kagglehub/handle.py:52]: nit: `has_version(self)`

Comment (mohami2000) [reply src/kagglehub/handle.py:52]: we use is_versioned to keep it consistent with the model handle

File: src/kagglehub/handle.py [hunk 2]
```diff
@@ -43,6 +43,28 @@ def to_url(self) -> str:
...
+
+    def __str__(self) -> str:
+        handle_str = f"{self.owner}/{self.dataset_name}"
+        if self.is_versioned():
+            return f"{handle_str}/{self.version}"
+        return handle_str
+
+    def to_url(self) -> str:
+        if self.is_versioned():
+            return f"https://www.kaggle.com/datasets/{self.owner}/{self.dataset_name}/versions/{self.version}"
```
Comment (Philmod) [inline src/kagglehub/handle.py:63]: You can probably simplify by reusing the base URL like you did for `__str___`: 

```
base_url = f"https://www.kaggle.com/datasets/{self.owner}/{self.dataset_name}"
if self.is_versioned():
  return f"{base_url}/versions/{self.version}"
return base_url
```

**Comment by neshdev** [comment]
LGTM. Its normally good practice to add unit tests along with the implementation, but this seems simple enough.

**Review by neshdev — APPROVED** [review]

**Pull request merged**